src/NodaTime/Calendars/Wondrous/WondrousCalendarHelper.cs

```diff
@@ -0,0 +1,2204 @@
+    {
+        private const int Month18 = 18;
+        private const int DaysInMonth = 19;
+
+
+        /// <remarks>
+        ///     There are 19 months in a year. Between the 18th and 19th month are the "days of Ha" (Ayyam-i-Ha).
+        ///     Options for numbering the months:
+        ///     - treat Ayyam-i-Ha as month 0. This will cause problems if 0 is treated as unknown. When stored internally,
+        ///       NodaTime months are converted to 0-based numbers, so this would by -1 which is not supported.
+        ///     - treat Ayyam-i-Ha as month -1. This would be a problem if the base classes convert to 0 based months
+        ///     - treat Ayyam-i-Ha as month 19. This is confusing because the 19th month would be displayed as month #20.
+        ///     - treat Ayyam-i-Ha as month 20. This will cause problems if months are sorted naively.
+        ///     - treat Ayyam-i-Ha as extra days in month 18. This would require special handling to sometimes display
+        ///       days in "month 18" as being Ayyam-i-Ha, but we have no direct control over display.
+        ///       
+        /// This version will treat Ayyam-i-Ha days as extra days in month 18.
+        /// 
+        /// Users interacting with the system should use 0 as Ayyam-i-Ha.
+        /// </remarks>
+        internal const int AyyamiHaMonth0 = 0;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="wondrousYear">Year in the Wondrous calendar</param>
+        /// <param name="wondrousMonth">Month (use 0 for Ayyam-i-Ha)</param>
+        /// <param name="wondrousDay">Day in month</param>
+        /// <returns></returns>
+        public static LocalDate CreateDate(int wondrousYear, int wondrousMonth, int wondrousDay)
+        {
+            if (wondrousMonth == AyyamiHaMonth0)
+            {
+                var maxDay = GetYearInfo(wondrousYear).DaysInAyyamiHa;
+
+                Preconditions.CheckArgumentRange(nameof(wondrousDay), wondrousDay, 1, maxDay);
+
+                // move Ayyam-i-Ha days to fall after the last day of month 18.
+                wondrousMonth = Month18;
+                wondrousDay += DaysInMonth;
+            }
+            return new LocalDate(wondrousYear, wondrousMonth, wondrousDay, CalendarSystem.Wondrous);
+        }
+
+        /// <summary>
+        /// Create a date in the Wondrous calendar.
+        /// </summary>
+        /// <param name="wondrousYear"></param>
+        /// <param name="wondrousMonth"></param>
+        /// <param name="wondrousDay"></param>
+        /// <param name="hour"></param>
+        /// <param name="minute"></param>
+        /// <param name="second"></param>
+        /// <param name="millisecond"></param>
+        /// <returns></returns>
+        public static LocalDateTime CreateDateTime(int wondrousYear, int wondrousMonth, int wondrousDay, int hour, int minute = 0, int second = 0, int millisecond = 0)
+        {
+            if (wondrousMonth == AyyamiHaMonth0)
```

Comment:
Is there any reason this method isn't implemented via `CreateDate(year, month, day) + new LocalTime(hour, minute, second,...)`?

Reply:
Not really, will do.

Reply:
Removed it entirely.